test/wrappers/ortools_test.rb

```diff
@@ -5770,4 +5770,31 @@ def test_self_selection_should_not_change_vehicle_ids
       end
     end
   end
+
+  def test_quantity_precision
+    problem = VRP.lat_lon_two_vehicles
+    problem[:services].each{ |service|
+      service[:quantities] = [{ unit_id: 'kg', value: 1.001 }]
+    }
+    problem[:vehicles].each{ |vehicle|
+      vehicle[:capacities] = [{ unit_id: 'kg', limit: 13 }]
+    }
+
+    result = OptimizerWrapper.wrapper_vrp('demo', { services: { vrp: [:ortools] }}, TestHelper.create(problem), nil)
+    assert_equal 13, result[:routes].map{ |route| route[:activities].count{ |activity| activity[:service_id] } }.sum,
+                 'The problem and result are expected to contain 13 services'
+
+    result[:routes].each{ |route|
+      assert_operator route[:activities].count{ |activity| activity[:service_id] }, :<=, 12,
+                      'Routes cannot load more than 12 services'
+      collected_quantities = route[:activities].map{ |activity|
+        next unless activity[:service_id]
+
+        assert_equal 1.001, activity[:detail][:quantities].first[:value]
+        activity[:detail][:quantities].first[:value]
+      }.compact.sum
+      assert_operator collected_quantities, :<, 13,
+                      'Routes cannot load more than 13 kg'
```

Comment:
last assert is not necessary it is a double of the first asserts. 

if there are at most 12 activities and all are equal to 1.001 then the sum cannot be more than 12 * 1.001 (which is always < 13)